spec/lib/bitmap_editor_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'bitmap_editor'
+
+RSpec.describe BitmapEditor do
+  context '.run' do
+    context 'with wrong file path' do
+      it 'returns an error message asking for a correct path' do
+        expect do
+          BitmapEditor.new.run('wrong/path/file.txt')
+        end.to output("Please provide correct file\n").to_stdout
+      end
+    end
+
+    context 'with empty file' do
+      it 'returns an error message indicating no command found' do
+        expect do
+          BitmapEditor.new.run('spec/fixtures/empty.txt')
+        end.to output("No command found\n").to_stdout
+      end
+    end
+
+    context 'with file showing no initialised bitmaps' do
+      it 'returns an error message indicating no command found' do
+        expect do
+          BitmapEditor.new.run('spec/fixtures/no_bitmap.txt')
+        end.to output("There is no image\n").to_stdout
+      end
+    end
+
+    context 'with file showing initial white bitmap' do
+      it 'shows a white bitmap of 3 x 2' do
+        expect do
+          BitmapEditor.new.run('spec/fixtures/init.txt')
+        end.to output("OOO\nOOO\n").to_stdout
+      end
+
+      it 'has an image of the right size' do
+        editor = BitmapEditor.new
+        editor.run('spec/fixtures/init.txt')
+        expect(editor.bitmap.width).to eq(3)
+        expect(editor.bitmap.height).to eq(2)
```

Comment:
there's a couple of things we could do to clean up this and the next test. if the setup is repetitious we should move that out (and try to use `let`). Also it's good practise to have only one expectation per `it`.
```
context 'with file showing initial white bitmap' do 
  subject { BitmapEditor.new.run(path).bitmap }

  context 'has an image of the right size'
   let(:path) { 'spec/fixtures/init.txt' }

   its(:width) { is_expected.to eq(3) }
   its(:height) { is_expected.to eq(2) }
  end
end
```